src/query/functions/temporal/base.go

```diff
@@ -127,47 +123,27 @@ func (c *baseNode) Process(
 		return fmt.Errorf("bound duration cannot be 0, bounds: %v", bounds)
 	}
 
-	seriesIter, err := b.SeriesIter()
-	if err != nil {
-		return err
-	}
-
-	// rename series to exclude their __name__ tag as part of function processing.
-	resultSeriesMeta := make([]block.SeriesMeta, 0, len(seriesIter.SeriesMeta()))
-	for _, m := range seriesIter.SeriesMeta() {
-		tags := m.Tags.WithoutName()
-		resultSeriesMeta = append(resultSeriesMeta, block.SeriesMeta{
-			Name: tags.ID(),
-			Tags: tags,
-		})
-	}
-
-	builder, err := c.controller.BlockBuilder(queryCtx, meta, resultSeriesMeta)
-	if err != nil {
-		return err
-	}
-
-	steps := bounds.Steps()
-	if err := builder.AddCols(steps); err != nil {
-		return err
-	}
-
 	m := blockMeta{
 		end:         xtime.ToUnixNano(bounds.Start),
-		seriesMeta:  resultSeriesMeta,
+		queryCtx:    queryCtx,
 		aggDuration: xtime.UnixNano(c.op.duration),
 		stepSize:    xtime.UnixNano(bounds.StepSize),
-		steps:       steps,
+		steps:       bounds.Steps(),
```

Comment:
Hm, any reason we're not excluding __name__ part any more? (as per prior comment `// rename series to exclude their __name__ tag as part of function processing.`)

Reply:
Still around, just moved some logic around; will readd this comment around the relevant paths :+1: